sdk/eventhub/event-hubs/src/util/connectionStringUtils.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { parseConnectionString } from "@azure/core-amqp";
+
+/**
+ * The set of properties that comprise an Event Hub connection string.
+ */
+export interface EventHubConnectionStringProperties {
+  /**
+   * The fully qualified Event Hub namespace extracted from the "Endpoint" in the
+   * connection string. This is likely to be similar to "{yournamespace}.servicebus.windows.net".
+   * This is typically used to construct the ServiceBusClient.
+   */
+  fullyQualifiedNamespace: string;
+  /**
+   * The value for "Endpoint" in the connection string.
+   */
+  endpoint: string;
+  /**
+   * The value for "EntityPath" in the connection string which would be the name of the queue or
+   * topic associated with the connection string.
+   * Connection string from a Shared Access Policy created at the namespace level
+   * will not have the EntityPath in it.
+   */
+  eventHubName?: string;
+  /**
+   * The value for "SharedAccessKey" in the connection string. This along with the "SharedAccessKeyName"
+   * in the connection string is used to generate a SharedAccessSignature which can be used authorize
+   * the connection to the service.
+   */
+  sharedAccessKey?: string;
+  /**
+   * The value for "SharedAccessKeyName" in the connection string. This along with the "SharedAccessKey"
+   * in the connection string is used to generate a SharedAccessSignature which can be used authorize
+   * the connection to the service.
+   */
+  sharedAccessKeyName?: string;
+  /**
+   * The value for "SharedAccessSignature" in the connection string. This is typically not present in the
+   * connection string generated for a Shared Access Policy. It is instead generated by the
+   * user and appended to the connection string for ease of use.
+   */
+  sharedAccessSignature?: string;
+}
+
+/**
+ * Parses given connection string into the different properties applicable to Azure Event Hubs.
+ * The properties are useful to then construct an EventHub{Producer|Consumer}Client.
+ * @param connectionString The connection string associated with the Shared Access Policy created
+ * for the Service Bus namespace, queue or topic.
+ */
+export function parseEventHubConnectionString(
+  connectionString: string
+): Readonly<EventHubConnectionStringProperties> {
```

Comment:
Just wanted to call out that this is different than `parseServiceBusConnectionString` now because it returns a `Readonly<T>` - it felt like a reasonable thing to do but feel free to let me know if I should revert that to match the behavior of service bus 😄 (or update `parseServiceBusConnectionString`)

Reply:
Did we want it to be a read-only, @ramya-rao-a ?

Reply:
Can you point to the reqs that needed this to be readonly? Am just curious as to what do we gain by making this readonly.

Reply:
Sure! If you look in the ticket #11894 there's an item that says `The parsed values should be treated as read-only, if idiomatic to the language.` - I know we used immutableJS quite a bit and immutability is definitely idiomatic in JS so it seemed like it lined up? Whether `Readonly` is idiomatic TS I honestly don't know, and whether we use `readonly` much in our repo is something else I didn't really know at the time.

Then I looked at the function here and that it parses a connection string. It felt right that the parsed result stays readonly so it always remains a true representation of the connection string that was used to parse it.

With all that said - it's really not a huge deal either way, so I'm happy to change it! but that was my reasoning at the time.

Reply:
If we do want to make these readonly, we should make use of [Object.defineProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) so that we make these fields truly readonly versus only readonly in the type system.

Reply:
IMO `readonly` is fine, I won't say its idiomatic in TS but it gets used a lot, and it's only a compile-time hint that costs pretty much nothing for us to implement.

Reply:
@chradek I don't think we gain much by making these fields actually read only, unless we feel like there's a strong risk that JS users would think mutating these properties does something.

Reply:
I don't see a scenario where JS users (or TS users for that matter) mutating these properties would do any harm. The only one reason I would see to be consistent with SB is for the future scenario where we will be sharing code. Even then, this final output is going to be named different anyway so if `<Readonly>` is added to one and not the other, there is not much of a problem